sha3/cshake.go

```diff
@@ -0,0 +1,81 @@
+package sha3
+
+// This allows you to customize SHAKE with a custom string,
+// see shake.go for more comments on usage and security.
+
+import (
+	"io"
+	"bytes"
+	"encoding/binary"
+)
+
+type CShakeHash interface {
+	io.Writer
+	io.Reader
+	Clone() ShakeHash
+	Reset()
+}
+
+// NewcShake128 creates a new custom cSHAKE128 variable-output-length CShakeHash.
```

Comment:
Can you split this cshake implementation into a separate commit?

Reply:
Sure